include/ensmallen_bits/moead/weight_init_policies/uniform_init.hpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @file das_dennis_init.hpp
+ * @author Nanubala Gnana Sai
+ *
+ * The Uniform(Das Dennis) methodology of Weight Initialization.
+ *
+ * ensmallen is free software; you may redistribute it and/or modify it under
+ * the terms of the 3-clause BSD license.  You should have received a copy of
+ * the 3-clause BSD license along with ensmallen.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef ENSMALLEN_MOEAD_UNIFORM_HPP
+#define ENSMALLEN_MOEAD_UNIFORM_HPP
+
+namespace ens {
+
+class Uniform
+{
+ public:
+  Uniform()
+  {
+    /* Nothing to do. */
+  }
+
+  template<typename MatType>
+  MatType Generate(size_t numObjectives, size_t numPoints, double epsilon)
+  {
+    size_t numPartitions  = FindNumParitions(numPoints, numObjectives);
+    size_t validNumPoints = FindNumUniformPoints(numPartitions, numObjectives);
+
+    //! The requested number of points is not matching any partition number.
+    if (numPoints != validNumPoints)
+    {
+      size_t nextValidNumPoints = FindNumUniformPoints(numPartitions + 1, numObjectives);
+      std::ostringstream oss;
+      oss << "DasDennis::Generate(): " << "The requested numPoints " << numPoints
+          << " cannot be generated uniformly.\n " << "Either choose numPoints as "
+          << validNumPoints << "(numPartition = " << numPartitions << ") or "
+          << "numPoints as " << nextValidNumPoints << "(numPartition = "
+          << numPartitions + 1 << ").";
+      throw std::logic_error(oss.str());
+    }
+
+    return DasDennis<MatType>(numPartitions, numObjectives,
+        numPoints, epsilon);
+  }
+
+ private:
+   /**
+    * Finds the number of points which can be sampled from the unit
+    *  simplex given the number of partitions.
+    */
+  size_t FindNumUniformPoints(size_t numPartitions, size_t numObjectives)
+  {
+    auto BinomialCoefficient =
+        [](size_t n, size_t k) -> double
+        {
+          return std::tgamma(n + 1) / (std::tgamma(k + 1) * std::tgamma(n - k + 1));
```

Comment:
I see, this is what @coatless suggested?

Reply:
No, this is a binomial coefficient. That one was Dirichilet